src/main/java/com/google/sps/servlets/ChallengesServlet.java

```diff
@@ -13,38 +13,111 @@
 // limitations under the License.
 
 package com.google.sps.servlets;
+import com.google.gson.*;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken.Payload;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdTokenVerifier;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.extensions.appengine.http.UrlFetchTransport;
+import com.google.api.client.json.gson.GsonFactory;
+import com.google.api.client.json.JsonFactory;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.QueryResultList;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.FilterOperator;
 
-import com.google.gson.Gson;
 import com.google.sps.data.Challenge;
 import com.google.sps.data.ChallengeData;
+import com.google.sps.data.GoogleIdHelper;
+import com.google.sps.data.User;
+
 import java.io.IOException;
 import java.net.URL;
 import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Collection;
+import java.util.Collections;
 import java.util.List;
 import java.util.Scanner;
+
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+import javafx.util.Pair;
+
 /** Servlet that manages challenges **/
 @WebServlet("/challenges")
 public class ChallengesServlet extends HttpServlet {
+  private static final int NO_CHALLENGES = 0;
+  private static final String NUM_CHALLENGES = "num-challenges";
+  private static final String ID_TOKEN = "id_token";
+  private ArrayList<Challenge> requested_challenge_list = new ArrayList<>();
 
-    @Override
-    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-      String json = convertToJsonUsingGson(ChallengeData.challenges);
-      response.setContentType("application/json;");
-      response.getWriter().println(json);
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    int num_challenges = getNumChallenges(request);
```

Comment:
`num_challenges` seems to never be used - is that intentional?

Reply:
Oh it is used in on line 163